### DISCUSSION TITLE
Trouble starting the server after transition from medusa-extender

### DISCUSSION THREAD

**Discussion by kjedras** [medusajs/medusa#4156] [Q&A]
Hey, im trying to migrate my marketplace from extender to pure medusa. When im trying to start the server im getting "Error starting server" during plugin initialization, thats pointing at _"registerPaymentServiceFromClass"_ function from `/loaders/helpers/plugin.js` file. I'm using latest _"medusa-payment-manual"_ plugin, i've tried commenting out every plugin but nothing changed. Any idea what could cause this problem?

Medusa v1.11
Error stack:
{
  "level": "error",
  "message": "Error starting server",
  "stack": [
    {
      "columnNumber": 51,
      "fileName": "node_modules\\@medusajs\\medusa\\dist\\loaders\\helpers\\plugins.js",
      "functionName": "registerPaymentServiceFromClass",
      "lineNumber": 8,
      "methodName": null,
      "native": false,
      "typeName": null
    },
    {
      "columnNumber": 87,
      "fileName": "node_modules\\@medusajs\\medusa\\dist\\loaders\\plugins.js",
      "functionName": null,
      "lineNumber": 363,
      "methodName": null,
      "native": false,
      "typeName": null
    },
    {
      "columnNumber": 23,
      "fileName": "node_modules\\@medusajs\\medusa\\dist\\loaders\\plugins.js",
      "functionName": "step",
      "lineNumber": 33,
      "methodName": null,
      "native": false,
      "typeName": null
    },
    {
      "columnNumber": 53,
      "fileName": "node_modules\\@medusajs\\medusa\\dist\\loaders\\plugins.js",
      "functionName": "Object.next",
      "lineNumber": 14,
      "methodName": "next",
      "native": false,
      "typeName": "Object"
    },
    {
      "columnNumber": 71,
      "fileName": "node_modules\\@medusajs\\medusa\\dist\\loaders\\plugins.js",
      "functionName": null,
      "lineNumber": 8,
      "methodName": null,
      "native": false,
      "typeName": null
    },
    {
      "columnNumber": null,
      "fileName": null,
      "functionName": "new Promise",
      "lineNumber": null,
      "methodName": null,
      "native": false,
      "typeName": null
    },
    {
      "columnNumber": 12,
      "fileName": "node_modules\\@medusajs\\medusa\\dist\\loaders\\plugins.js",
      "functionName": "__awaiter",
      "lineNumber": 4,
      "methodName": null,
      "native": false,
      "typeName": null
    },
    {
      "columnNumber": 87,
      "fileName": "node_modules\\@medusajs\\medusa\\dist\\loaders\\plugins.js",
      "functionName": null,
      "lineNumber": 354,
      "methodName": null,
      "native": false,
      "typeName": null
    },
    {
      "columnNumber": null,
      "fileName": null,
      "functionName": "Array.map",
      "lineNumber": null,
      "methodName": "map",
      "native": false,
      "typeName": "Array"
    },
    {
      "columnNumber": 60,
      "fileName": "node_modules\\@medusajs\\medusa\\dist\\loaders\\plugins.js",
      "functionName": null,
      "lineNumber": 354,
      "methodName": null,
      "native": false,
      "typeName": null
    }
  ],
  "timestamp": "2023-05-23 12:19:36"
}

**Comment by olivermrbl** [comment]
@kjedras - please open this as an issue and add the requested details in the template :)

**Comment by nickngqs** [comment]
I'm getting this error too, can't really figure out why.

**Comment by SGFGOV** [comment]
Just a thought have you

1. Exported a default from the service class?
2. Created migrations and successfully executed them

On Fri, 2 Jun, 2023, 06:22 nickngqs, ***@***.***> wrote:

> I'm getting this error too, can't really figure out why.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/medusajs/medusa/discussions/4156#discussioncomment-6066542>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXEQJHEBL26WU62GEQ6Q7QDXJE2OTANCNFSM6AAAAAAYLV5ZIQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

-- 

This message (including any attachments) may contain confidential, 
proprietary, privileged and/or private
information. The information is 
intended to be for the use of the individual or entity designated above. If
you are not the intended recipient of this message, please notify the 
sender immediately, and delete the
message and any attachments. Any 
disclosure, reproduction, distribution or other use of this message or
any 
attachments by an individual or entity other than the intended recipient is 
prohibited.
